pkg/kube/ready.go

```diff
@@ -76,10 +76,11 @@ func NewReadyChecker(cl kubernetes.Interface, log func(string, ...interface{}),
 
 // ReadyChecker is a type that can check core Kubernetes types for readiness.
 type ReadyChecker struct {
-	client        kubernetes.Interface
-	log           func(string, ...interface{})
-	checkJobs     bool
-	pausedAsReady bool
+	client         kubernetes.Interface
+	log            func(string, ...interface{})
+	checkJobs      bool
+	pausedAsReady  bool
+	readyResources []*resource.Info
```

Comment:
Used for deduplication

Reply:
I think a map would be better here, it gets you de-duplication which looks like it would be a problem since the check loop PollUntilContextCancel will start over on failures. If you have 1000 items its going to grow by at least 999 each loop. 
There is also prior art for this and I would use the same key linked below so if in the future this map needs to be dumped it's clear what objects are in there. 
GVK/ns/name is a common pattern to keep track of objects in k8s as it's guaranteed unique across the cluster. 
https://github.com/helm/helm/blob/b5e55bf21039a371437ee859dfb704119748f797/pkg/action/upgrade.go#L319
https://github.com/helm/helm/blob/b5e55bf21039a371437ee859dfb704119748f797/pkg/action/upgrade.go#L604